### PR TITLE
DRAFT Generators

### PR DESCRIPTION
**Pull request by clavedeluna**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [ ] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -86,4 +86,4 @@ Tests are in the 'test' subdirectory. To launch the whole tests suite, you can u
 either `tox` or `pytest`::
 
     tox
-    pytest astroid
+    pytest tests
```
Comment (DanielNoord) [inline README.rst:89]: This should probably be done in a separate PR.

**Comment by DanielNoord** [comment]
@clavedeluna I have added some comments and some clean up to get you going! This should set you up to implement at least `__enter__`. `__exit__` might be a bit harder, but follows a similar pattern.

File: tests/unittest_object_model.py [hunk 2]
```diff
@@ -605,6 +607,14 @@ def test():
         send = next(ast_nodes[4].infer())
         self.assertIsInstance(send, astroid.BoundMethod)
 
+        enter = next(ast_nodes[5].infer())
+        assert isinstance(enter, astroid.BoundMethod)
+        # TODO: Some test about the args to be passed to this method
```
Comment (DanielNoord) [inline tests/unittest_object_model.py:612]: Oh, afterthought: this should also test the return value!

**Comment by DanielNoord** [comment]
Oops, sorry! Didn't mean to mark as non-draft (meant to run the tests).  I can't seem to undo this on mobile 😅

**Comment by clavedeluna** [comment]
@DanielNoord I spent some time on this but it's over my head at this time and more than I'm able to do. I'll focus on helping out in other areas in need so feel free to pick this up if it suits you.

**Comment by DanielNoord** [comment]
Closing in favour of https://github.com/PyCQA/astroid/pull/1810.
